build/devenv/tests/e2e/smoke_test.go

```diff
@@ -703,3 +691,118 @@ func multiVerifierTestCases(t *testing.T, src, dest uint64, in *ccv.Cfg, c *evm.
 		},
 	}
 }
+
+// ============================================================================
+// RMN Curse Helper Functions
+// ============================================================================
+
+// chainSelectorToSubject converts a chain selector to a bytes16 curse subject.
+func chainSelectorToSubject(chainSel uint64) [16]byte {
+	var result [16]byte
+	// Convert the uint64 to bytes and place it in the last 8 bytes of the array
+	binary.BigEndian.PutUint64(result[8:], chainSel)
+	return result
+}
+
+// globalCurseSubject returns the global curse constant.
+func globalCurseSubject() [16]byte {
+	return globals.GlobalCurseSubject()
+}
+
+// ============================================================================
+// RMN Curse Tests
+// ============================================================================
+
+func TestRMNCurseLaneVerifierSide(t *testing.T) {
+	testCtx, selectors := NewDefaultTestingContext(t, "../../env-out.toml", 3)
+	c := testCtx.Impl
+	ctx := testCtx.Ctx
+	l := zerolog.Ctx(ctx)
+
+	var err error
+	chain0, chain1, chain2 := selectors[0], selectors[1], selectors[2]
+	receiver := mustGetEOAReceiverAddress(t, c, chain1)
+
+	sentEvt := testCtx.MustSendMessage(chain0, chain1, receiver, 150) // Use custom finality to slow down picking for verification
+	messageID := sentEvt.MessageID
+
+	l.Info().Msg("Applying lane curse between chain0 and chain1 (before message gets picked up by verifier)")
+	// normally it's bidirectional, for the sake of the test we only curse one direction
+	err = c.ApplyCurse(ctx, chain0, [][16]byte{chainSelectorToSubject(chain1)})
+	require.NoError(t, err)
+
+	l.Info().Msg("Asserting baseline message reaches verifier but gets dropped due to curse")
+	testCtx.AssertMessageReachedAndDroppedInVerifier(messageID, 100*time.Second)
```

Comment:
Feels like we need to come up with a better way to assert things like this more than ever, since this is a pretty long timeout (longer than half the other tests in TestSmoke combined)